server/service/software.go

```diff
@@ -0,0 +1,40 @@
+package service
```

Comment:
Should the filename be called `endpoint_software.go` or am I misunderstanding the existing convention?

Reply:
The idea is that we stop having `endpoint_something.go`, `transport_something.go`, and `service_something.go` and we just have `something.go` so all that goes to whatever "something" is in the same file. Most of the complexity will live in the datastore layer or the service, so this file should mostly be a set of service methods that rely a lot on the datastore.

It's the goal that doing this will be faster to read (instead of having to jump around many files) and also should keep us in check if suddenly adding new things becomes more verbose, we should think about simplifying.

All the files that are named `something.go` instead of the old convention, also use the new handler declaration, which is shorter, and the generic decoder. I feel like it's an improvement, but I think it's just one of the few steps we'll take to make adding these things faster and less error prone.

That's basically the idea, what do you think?

Reply:
Gotcha, yeah makes sense to me, I agree that it makes it easier to read when all in one file.